commands/message.go

```diff
@@ -221,6 +223,82 @@ var msgWaitCmd = &cmds.Command{
 	},
 }
 
+// MessageStatusResult is the status of a message on chain or in the message queue/pool
+type MessageStatusResult struct {
+	InPool    bool
+	PoolMsg   *types.SignedMessage
+	InOutbox  bool
+	OutboxMsg *core.QueuedMessage
+	OnChain   bool
+	ChainMsg  *msg.ChainMessage
+}
+
+var msgStatusCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show status of a message",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, false, "CID of the message to inspect"),
+	},
+	Options: []cmdkit.Option{},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		msgCid, err := cid.Parse(req.Arguments[0])
+		if err != nil {
+			return errors.Wrap(err, "invalid cid "+req.Arguments[0])
+		}
+
+		api := GetPorcelainAPI(env)
+		result := MessageStatusResult{}
+
+		// Look in message pool
+		result.PoolMsg, result.InPool = api.MessagePoolGet(msgCid)
+
+		// Look in outbox
+		for _, addr := range api.OutboxQueues() {
+			for _, qm := range api.OutboxQueueLs(addr) {
+				cid, err := qm.Msg.Cid()
+				if err != nil {
+					return err
+				}
+				if cid.Equals(msgCid) {
+					result.InOutbox = true
+					result.OutboxMsg = qm
+				}
+			}
+		}
+
+		// Look on chain
+		result.ChainMsg, result.OnChain, err = api.MessageFind(req.Context, msgCid)
+		if err != nil {
+			return err
+		}
+		return re.Emit(&result)
+	},
+	Type: &MessageStatusResult{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, res *MessageStatusResult) error {
```

Comment:
i would have expected printing the message out below as well

Reply:
It does at line 294